Header header
float32[] vis
VisNode[] crit_points
